lib/base/connection.js

```diff
@@ -0,0 +1,556 @@
+          Object.assign(config.pool, {
+            min: value
+          })
+          break
+        case 'multipleactiveresultsets':
+          break
+        case 'multisubnetfailover':
+          Object.assign(config.options, {
+            multiSubnetFailover: value
+          })
+          break
+        case 'network library':
+          break
+        case 'packet size':
+          Object.assign(config.options, {
+            packetSize: value
+          })
+          break
+        case 'password':
+          Object.assign(config, {
+            password: value
+          })
+          break
+        case 'persist security info':
+          break
+        case 'poolblockingperiod':
+          break
+        case 'pooling':
+          break
+        case 'replication':
+          break
+        case 'transaction binding':
+          Object.assign(config.options, {
+            enableImplicitTransactions: value.toLowerCase() === 'implicit unbind'
+          })
+          break
+        case 'transparentnetworkipresolution':
+          break
+        case 'trustservercertificate':
+          Object.assign(config.options, {
+            trustServerCertificate: value
+          })
+          break
+        case 'type system version':
+          break
+        case 'user id': {
+          let user = value
+          let domain
+          if (/^(.*)\\(.*)$/.exec(user)) {
+            domain = RegExp.$1
+            user = RegExp.$2
+          }
+          Object.assign(config, {
+            domain,
+            user
+          })
+          break
+        }
+        case 'user instance':
+          break
+        case 'workstation id':
+          Object.assign(config.options, {
+            workstationId: value
+          })
+          break
+        case 'request timeout':
+          Object.assign(config, {
+            requestTimeout: parseInt(value, 10)
+          })
+          break
+        case 'stream':
+          Object.assign(config, {
+            stream: !!value
+          })
+          break
+        case 'useutc':
+          Object.assign(config.options, {
+            useUTC: !!value
+          })
+          break
+        case 'parsejson':
+          Object.assign(config, {
+            parseJSON: !!value
+          })
+          break
+      }
+      return config
+    }, { options: {}, pool: {} })
+  }
+
+  get connected() {
+    return this.opened
+  }
+
+  get opened() {
+    return !!this._connection
+  }
+
+  get inTransaction() {
+    return this._activeInstance instanceof shared.driver.Transaction;
+  }
+
+  /**
+   * Creates a new connection pool with one active connection. This one initial connection serves as a probe to find out whether the configuration is valid.
+   *
+   * @param {basicCallback} [callback] A callback which is called after connection has established, or an error has occurred. If omited, method returns Promise.
+   * @return {ConnectionPool|Promise}
+   */
+
+  async open() {
```

Comment:
the doc block says a callback can be provided, however it is not implemented.

I also don't think we should use `async` as we are not using that anywhere else in the lib and it means we can't return the same `Promise` type that has been injected by the developer.